### DISCUSSION TITLE
Route caching does not work when adding channels route via package

### DISCUSSION THREAD

**Discussion by dominik-eller** [spatie/laravel-package-tools#126] [Q&A]
I have a package and I am adding routes like that:

    ->hasRoute('web')
    ->hasRoute('channels')

my routes are placed in the following directory /routes and everything is working as expected until I cache the routes using the following command:

    php artisan route:cache

after running that my custom channels.php is missing when executing:

    php artisan channel:list

Running php artisan route:clear solves it as well as adding the following line to die BroadcastServiceProvider

    require base_path('vendor/***/***/routes/channels.php');

So I guess in the default route publishing something is missing for channels.php route file.

**Comment by lao9s** [comment]
+1

**Comment by SlimDeluxe** [comment]
Any solution found for this?
